WordPress/src/test/java/org/wordpress/android/ui/posts/PrepublishingHomeViewModelTest.kt

```diff
@@ -211,8 +216,29 @@ class PrepublishingHomeViewModelTest : BaseUnitTest() {
         assertThat(headerUiState?.siteName?.text).isEqualTo(expectedName)
     }
 
+    @Test
+    fun `verify that tapping publish button will invoke onPublishButtonClicked`() {
+        // arrange
+        var event: Event<Unit>? = null
+        viewModel.onPublishButtonClicked.observeForever {
+            event = it
+        }
```

Comment:
I remember you mentioned this approach previously. I did it because it seems to be best approach for events that don't contain a value. Let me know if it can be simplified 🙏

Reply:
I'm not sure I follow. Even events which don't contain any value will receive the event - the event will just be of type `Unit`. In other words we can still invoke `getContentIfNotHandled` and check if we received the event after the button was clicked right?

Having said that, no need to change the code. Both approaches are valid - I think it's a matter of personal preference and in this case both are equally readable.

Reply:
Okay thanks for the explanation! I never thought about the `getContentIfNotHandled` in that way since I was always using the `applyIfNotHandled` variant with `Unit` events, it never clicked in my mind that I could just get the event haha 😅